homeassistant/components/overkiz/climate_entities/atlantic_electrical_heater_with_adjustable_temperature_setpoint.py

```diff
@@ -0,0 +1,138 @@
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.COMFORT_1: PRESET_COMFORT1,
+    OverkizCommandParam.COMFORT_2: PRESET_COMFORT2,
+    OverkizCommandParam.AUTO: PRESET_AUTO,
+    OverkizCommandParam.BOOST: PRESET_BOOST,
+    OverkizCommandParam.INTERNAL: PRESET_PROG,
+}
+
+PRESET_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODE.items()}
+
+# Map Overkiz HVAC modes to Home Assistant HVAC modes
+OVERKIZ_TO_HVAC_MODE = {
+    OverkizCommandParam.ON: HVACMode.HEAT,
+    OverkizCommandParam.OFF: HVACMode.OFF,
+    OverkizCommandParam.AUTO: HVACMode.AUTO,
+    OverkizCommandParam.BASIC: HVACMode.HEAT,
+    OverkizCommandParam.STANDBY: HVACMode.OFF,
+    OverkizCommandParam.INTERNAL: HVACMode.AUTO,
+}
+
+HVAC_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODE.items()}
+
+
+class AtlanticElectricalHeaterWithAdjustableTemperatureSetpoint(
+    OverkizEntity, ClimateEntity
+):
+    """Representation of Atlantic Electrical Heater (With Adjustable Temperature Setpoint)."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODE_TO_OVERKIZ]
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str | None:
+        """Return hvac operation ie. heat, cool mode."""
+        if OverkizState.CORE_OPERATING_MODE in self.device.states:
+            state = self.executor.select_state(OverkizState.CORE_OPERATING_MODE)
+            return OVERKIZ_TO_HVAC_MODE[OverkizCommandParam(state)]
+        if OverkizState.CORE_ON_OFF in self.device.states:
+            state = self.executor.select_state(OverkizState.CORE_ON_OFF)
+            return OVERKIZ_TO_HVAC_MODE[OverkizCommandParam(state)]
+
+        return None
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        if OverkizState.CORE_OPERATING_MODE in self.device.states:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_OPERATING_MODE, HVAC_MODE_TO_OVERKIZ[hvac_mode]
+            )
+        elif hvac_mode == HVACMode.OFF:
+            await self.executor.async_execute_command(OverkizCommand.OFF)
+        else:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_HEATING_LEVEL, OverkizCommandParam.COMFORT
+            )
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if state := self.device.states[OverkizState.IO_TARGET_HEATING_LEVEL]:
```

Comment:
Well I just backported the code from ha-tahoma :S

Reply:
https://github.com/iMicknl/ha-tahoma/blob/69f9a2d91f4d6538364ae541628b3879709e9b7f/custom_components/tahoma/climate_devices/atlantic_electrical_heater_with_adjustable_temperature_setpoint.py#L201-L206

I think you added this in this PR? Since I don't have it in my closed version, and not on ha-tahoma. :)

Reply:
Here why I added this condition:

`homeassistant/components/overkiz/climate_entities/atlantic_electrical_heater_with_adjustable_temperature_setpoint.py:105: error: Item "None" of "Optional[State]" has no attribute "value"  [union-attr]`